src/connect.ts

```diff
@@ -110,6 +111,9 @@ export class ConnectQOS {
     if (hostStatus === ActorStatus.Bad) return BadActorType.badHost;
     else if (ipStatus === ActorStatus.Bad) return BadActorType.badIp;
 
+    // when we track host ratios, we block at the SOURCE IP
+    if (this.metrics.hostRatioViolations.has(host)) return BadActorType.badIp;
```

Comment:
I kinda like the idea of introducing a new `BadActorType` - would be nice to have a way to distinguish these from normal IP blocking so that we could log it differently on the DPS side. Since we're no longer blocking hosts we have lost the ability to see what host is getting slammed, and this would give us visibility into that. Thoughts?